### ISSUE TITLE
Documentation issues

### ISSUE THREAD

**Issue by Anteproperispomenon** [Plutonomicon/plutarch#468]
There's a few issues with the documentation:

In https://github.com/Plutonomicon/plutarch/blob/master/docs/Typeclasses/PConstant%20and%20PLift.md, the `DerivePConstantViaData` example just uses the same code as the `DerivePConstantViaNewtype` example.

In https://github.com/Plutonomicon/plutarch/blob/master/docs/Typeclasses/PIsDataRepr%20and%20PDataFields.md, the `PlutusTx.makeIsDataIndexed` uses `''FourWheeler` for the Type name instead of `Vehicle`.

If I see some other issues, I'll add them.

**Comment by groscoe** [comment]
Also, in https://github.com/Plutonomicon/plutarch/blob/master/docs/Typeclasses/PIsDataRepr%20and%20PDataFields.md, the `foo` example doesn't typecheck. I believe there's a `$` missing:

``` hs
foo :: Term s (PScriptContext :--> PString)
foo = plam $ \ctx -> P.do
  purpose <- pmatch $ pfield @"purpose" # ctx -- note the '$'
  case purpose of
    PMinting _ -> "It's minting!"
    PSpending _ -> "It's spending!"
    PRewarding _ -> "It's rewarding!"
    PCertifying _ -> "It's certifying!"
```

**Comment by TotallyNotChase** [comment]
thanks for letting us know! fixed in staging via #470
